source/develop/release-management/features/network/manageiq_ovn.html.md

```diff
@@ -0,0 +1,140 @@
+
+## Benefit to ManageIq
+* Adding more power to the oVirt's infra manager.
+* oVirt users that want to use OVN provider (more than the basic operations oVirt supports) will have to use ManageIQ.
+
+## Gaps with the OpenStack network provider
+
+### Gaps
+* Communication and authentication
+  * The OpenStack infra manager added to ManageIQ has the keystone URL. OpenStack infra manager communicates with the other OpenStack's services via the keystone. The keystone returns a service catalog with all the URLs and other information about the services. This information is used to communicate with the other services.
+  * The network manager is an ExtManagementSystem.
+  * OpenStack's network manager delegates all its communication and authentication methods to the parent manager (infra manager => keystone).
+  * The oVirt's infra manager is communicating with oVirt but the network manager should communicate directly with OVN provider. Therefore, the delegation the OpenStack's network manager has is problematic for oVirt.
+* Service Catalog
+  * In oVirt when adding a new network provider, the user supplies two values - `provider_url` and `authentication_url` (the URL of 'keystone').
+First an authentication is done against 'keystone' to get the `token` and then oVirt communicates with the provider using the `provider_url` (supplying the `token` for authentication).
+  * In ManageIQ the 'services' URLs are not supplied by the user but determined from the 'keystone'. When 'Keystone' is asked for the `token` it replies with the `token` and a `service_catalog` with all the services information (URLs, credentials, etc).
+* Tenants
+  * Neutron supports tenants.
+  * oVirt's OVN provider doesn't.
+  * ManageIQ expects the Neutron network provider to have tenants.
+* oVirt's OVN provider has 'v2' Neutron like API. However, 'v2' API is already considered as legacy.
+
+
+### Solving the gaps
+* Communication and authentication
+ * Creating network manager class that doesn't inherit from the OpenStack's one.
+ * oVirt's network manager will take response for its communication and authentication reusing the communication and authentication code of the OpenStack infra manager.
+ * The collector, parser, refresher and entities classes (cloud_network, cloud_subnet, network_port etc) inherit from the OpenStack ones.
+ * Upon each update of the oVirt's infra manager
+   * all the network providers will be retrieved from oVirt.
+   * only the first network provider will be processed.
+   * if a network manager doesn't exist it will be created.
+   * the network's provider `authentication_url` will be parsed and the network manager will be updated accordingly. The fields that will be updated are- `hostname`, `port`, `api_version` and `security_protocol` ('ssl' for 'https' and 'non-ssl' for 'http').
```

Comment:
dash-attached to a word is not a valid English punctuation. Use colon.

The fields that will be updated are:

Reply:
done